src/domains/setDomain.ml

```diff
@@ -366,3 +384,43 @@ struct
   let top () = of_list Elems.elems
   let is_top x = equal x (top ())
 end
+
+(** Set abstracted by a single (joined) element. *)
+module Joined (E: Lattice.S): S with type elt = E.t =
+struct
+  type elt = E.t
+  include E
+
+  let singleton e = e
+  let of_list es = List.fold_left E.join (E.bot ()) es
+  let exists p e = p e
+  let for_all p e = p e
+  let mem e e' = E.leq e e'
+  let choose e = e
+  let elements e = [e]
+  let remove e e' =
+    if E.leq e' e then
+      E.bot ()
+    else
+      e'
```

Comment:
We should also have comments warnings about these here

Reply:
I documented this both in the `SetDomain` interface and the specific implementations.